src/index.ts

```diff
@@ -1,174 +1,160 @@
+    ic_caller_id: ic_caller_id,
+    created_at: ic.time(),
+    updated_at: Opt.None,
+    ...payload
+  };
+  movieStorage.insert(movie.id, movie);
+  return Result.Ok(movie);
 }
 
-// Now lets create a function that can retrive all the movies stored in our canister
 $query;
 export function getAllMovies(): Result<Vec<Movie>, string> {
-    return Result.Ok(movieStorage.values());
+  return Result.Ok(movieStorage.values());
 }
 
-//Let us now create a function to retrive a specific movie using its unique identifier (ID)
 $query;
 export function getOneMovie(id: string): Result<Movie, string> {
-    return match(movieStorage.get(id), {
-        Some: (movie) => Result.Ok<Movie, string>(movie),
-        None: () => Result.Err<Movie, string>(`A movie with the ID: ${id} was not found. Please check the ID and try again.`)
-    });
+  return match(movieStorage.get(id), {
+    Some: (movie) => Result.Ok<Movie, string>(movie),
+    None: () => Result.Err<Movie, string>(`A movie with the ID: ${id} was not found. Please check the ID and try again.`)
+  });
 }
 
-// Now lets create a function that can allow us to update the details of a movie that already exists in our canister
 $update;
 export function updateMovie(id: string, payload: MoviePayload): Result<Movie, string> {
-    return match(movieStorage.get(id), {
-        Some: (movie) => {
-            const updatedMessage: Movie = {...movie, ...payload, updated_at: Opt.Some(ic.time())};
-            movieStorage.insert(movie.id, updatedMessage);
-            return Result.Ok<Movie, string>(updatedMessage);
-        },
-        None: () => Result.Err<Movie, string>(`Couldn't update a movie with the ID: ${id} because the movie was not found. Please check the ID and then try again.`)
-    });
+  return match(movieStorage.get(id), {
+    Some: (movie) => {
+      const updatedMessage: Movie = { ...movie, ...payload, updated_at: Opt.Some(ic.time()) };
+      movieStorage.insert(movie.id, updatedMessage);
+      return Result.Ok<Movie, string>(updatedMessage);
+    },
+    None: () => Result.Err<Movie, string>(`Couldn't update a movie with the ID: ${id} because the movie was not found. Please check the ID and then try again.`)
+  });
 }
 
-//The next step is to create a function that can help us remove a movie from our list
 $update;
 export function removeLoggedMovie(id: string): Result<Movie, string> {
-    return match(movieStorage.remove(id), {
-        Some: (removedMovie) => Result.Ok<Movie, string>(removedMovie),
-        None: () => Result.Err<Movie, string>(`Couldn't delete a movie with the ID: ${id} because the movie was not found. Please check the ID and then try again.`)
-    });
-}
-
-// a workaround to make uuid package work with Azle
-globalThis.crypto = {
-        getRandomValues: () => {
-        let array = new Uint8Array(32)
-
-        for (let i = 0; i < array.length; i++) {
-            array[i] = Math.floor(Math.random() * 256)
-        }
-
-        return array
-    }
```

Comment:
This custom method is an essential workaround to make the uuid package work with Azle. Thus removing it is not a good idea.